pkg/services/baremetal/host/state_machine.go

```diff
@@ -266,8 +267,31 @@ func (hsm *hostStateMachine) handleRegistering(ctx context.Context) actionResult
 
 	actResult := hsm.reconciler.actionRegistering(ctx)
 	if _, ok := actResult.(actionComplete); ok {
+		hsm.nextState = infrav1.StatePreProvisioning
+	}
+	return actResult
+}
+
+func (hsm *hostStateMachine) handlePreProvisioning(ctx context.Context) actionResult {
+	if hsm.provisioningCancelled() {
+		hsm.nextState = infrav1.StateDeprovisioning
+		return actionComplete{}
+	}
+
+	actResult := hsm.reconciler.actionPreProvisioning(ctx)
+	switch actResult.(type) {
+	case actionComplete:
 		hsm.nextState = infrav1.StateImageInstalling
+	case actionError:
+		// re-enable rescue system. If installimage failed, then it is likely, that
```

Comment:
this is the message we had in handleInstallImage.. I think it at least doesn't fit 100%

Reply:
fixed.